src/screens/Login/components/SignUpForm/index.js

```diff
@@ -1,29 +1,85 @@
-import React from 'react';
+import React, { useCallback } from 'react';
 import { string, func } from 'prop-types';
+import { useForm } from 'react-hook-form';
 import i18next from 'i18next';
 import Input from '@components/Input';
+import { ERROR_MESSAGES } from '@constants/errorMessages';
+import { VALIDATION_SCHEMA } from '../../constants';
 import styles from './styles.module.scss';
 
 const SignUpForm = ({ className, onChangeView }) => {
+  const {
+    register,
+    handleSubmit,
+    errors,
+    formState,
+    getValues,
+    clearError,
+    setError,
+  } = useForm();
+  const { dirtyFields } = formState;
+
+  const onSubmit = (data) => console.table(data);
+
+  const handleChangePasswordRepeat = useCallback(
+    (e) => {
+      const { password } = getValues();
+      const value = e.target.value;
+      if (value === password) return clearError('passwordRepeat');
+      setError('passwordRepeat', 'notMatch', ERROR_MESSAGES.passwordRepeat);
+    },
+    [clearError, getValues, setError]
+  );
+
   return (
-    <form className={className}>
+    <form className={className} onSubmit={handleSubmit(onSubmit)}>
       <h2 className='title m-bottom-4'>{i18next.t('LOGIN:TITLE')}</h2>
       <p className='base-text m-bottom-4'>{i18next.t('LOGIN:SUBTITLE')}</p>
-      <Input id='name' label={i18next.t('LOGIN:NAME')} />
-      <Input id='username' label={i18next.t('LOGIN:USER_NAME')} />
-      <Input id='email' label={i18next.t('LOGIN:EMAIL')} />
+      <Input
+        id='name'
+        label={i18next.t('LOGIN:NAME')}
+        register={register}
+        validationSchema={VALIDATION_SCHEMA.NAME}
+        errors={errors.name}
+        isDirty={dirtyFields.has('name')}
```

Comment:
I'd extract the field names to constants, specially because you're using them more than once